File: fixtures/scale/lua/large/checks/check_windmill.lua
local windmill = require("ciphers.windmill")
local windmill_spec = require("specs.windmill_spec")
local corpus_rotate = require("corpus.corpus_rotate")

local function check_windmill()
  local spec = windmill_spec.get()
  for _, text in ipairs(corpus_rotate.corpus_rotate()) do
    local sealed = windmill.windmill_encrypt(text, spec.key)
    local opened = windmill.windmill_decrypt(sealed, spec.key)
    if opened ~= text then return false end
  end
  return spec.category == "rotate"
end

return { check_windmill = check_windmill }

File: fixtures/scale/javascript/large/src/checks/checkWindmill.js
import windmill from '../ciphers/windmill.js';
import windmillSpecMod from '../specs/windmillSpec.js';
import corpusRotateMod from '../corpus/corpusRotate.js';

function checkWindmill() {
  const spec = windmillSpecMod.windmillSpec();
  for (const text of corpusRotateMod.corpusRotate()) {
    const sealed = windmill.windmillEncrypt(text, spec.key);
    const opened = windmill.windmillDecrypt(sealed, spec.key);
    if (opened !== text) return false;
  }
  return spec.category === 'rotate';
}

export default { checkWindmill };
